sshd-core/src/main/java/org/apache/sshd/server/forward/TcpipServerChannel.java

```diff
@@ -107,6 +110,22 @@ public TcpipServerChannel(ForwardingFilter.Type type) {
         return type;
     }
 
+    public SshdSocketAddress getTunnelEntrance() {
+        return tunnelEntrance;
+    }
+
+    public SshdSocketAddress getTunnelExit() {
+        return tunnelExit;
+    }
```

Comment:
Should be made as some sort of "marker" interface since implemented by several entities:
```java
public interface TunnelingChannelEndpointsProvider {
    SshdSocketAddress getTunnelEntrance();

    SshdSocketAddress getTunnelExit();
}
```

Reply:
I agree, I will shortly commit the new code with your right suggestion